src/Components/3DV/SceneView.tsx

```diff
@@ -224,8 +225,53 @@ function SceneView(props: ISceneViewProps, ref) {
     debugLog('debug', 'debug', 'SceneView Render');
     const url = modelUrl === 'Globe' ? globeUrl : modelUrl;
 
-    const sortMeshesOnLoad = () => {
+    const preProcessMeshesOnLoad = () => {
+        let uniqueNumber = 0;
+        //Initial loop to check for issues and cleanup meshes.
         for (const mesh of sceneRef.current.meshes) {
+            //Let's make sure that all mesh ids are unique
+            const matchingId = sceneRef.current.meshes.filter(
+                (x) => x.id == mesh.id
+            );
+            if (matchingId.length > 1) {
+                //Throw an error here
+                console.warn(
+                    'Loaded model contains objects with matching names. 3D Scenes Studio only supports 3d models with unique object names. Attempting to recover by forcing unique names...'
+                );
+                //Append unique numbers to the ids and move on
+                for (let i = 0; i < matchingId.length; i++) {
+                    matchingId[i].id += i;
+                }
+            }
+
+            //If the mesh is an InstancedMesh, break the mesh instancing to handle it as an independent object
+            if (mesh.isAnInstance) {
+                debugLog('debug', 'Breaking mesh instance: ', mesh.name);
+                const instancedMesh = mesh as BABYLON.InstancedMesh;
+                const deInstancedMesh = instancedMesh.sourceMesh.clone(
+                    instancedMesh.name + uniqueNumber,
+                    instancedMesh.parent
+                );
+                uniqueNumber++;
+                sceneRef.current.addMesh(deInstancedMesh);
+                console.log(deInstancedMesh.id);
+                sceneRef.current.removeMesh(deInstancedMesh);
+                instancedMesh.dispose();
+            }
+        }
+
+        //Loop again with the cleaned up mesh list
```

Comment:
do we need to do it "again"? or could we just wait to do it till we have done the processing?

Reply:
Though, I'm not seeing where we already looped